providers/asrockrack/asrockrack.go

```diff
@@ -0,0 +1,310 @@
+package asrockrack
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/bmc-toolbox/bmclib/internal/httpclient"
+	"github.com/bmc-toolbox/bmclib/providers"
+	"github.com/go-logr/logr"
+	"github.com/jacobweinstock/registrar"
+)
+
+const (
+	// BmcType defines the bmc model that is supported by this package
+	BmcType = "asrockrack"
+	// VendorID represents the id of the vendor across all packages
+	VendorID = "ASRockRack"
+	// ProviderName for the provider implementation
+	ProviderName = "asrockrack"
+	// ProviderProtocol for the provider implementation
+	ProviderProtocol = "vendorapi"
+)
+
+var (
+	// Features implemented by asrockrack https
+	Features = registrar.Features{
+		providers.FeatureBiosVersionRead,
+		providers.FeatureBmcVersionRead,
+		providers.FeatureBiosFirmwareUpdate,
+		providers.FeatureBmcFirmwareUpdate,
+	}
+)
+
+// ASRockRack holds the status and properties of a connection to a asrockrack bmc
+type ASRockRack struct {
+	ip            string
+	username      string
+	password      string
+	loginSession  *loginSession
+	httpClient    *http.Client
+	fwInfo        *firmwareInfo
+	resetRequired bool // Indicates if the BMC requires a reset
+	skipLogout    bool // A Close() / httpsLogout() request is ignored if the BMC was just flashed - since the sessions are terminated either way
+	ctx           context.Context
```

Comment:
this doesn't seem to be used anywhere. might consider removing it.

Reply:
All of these are in use..

Reply:
sorry, meant to say that I don't find any references to `ASRockRack.ctx` anywhere.

Reply:
thanks, I should have figured, fixed in bb18bfe4b7